packages/aws-rfdk/lib/lambdas/nodejs/lib/custom-resource/simple-resource.ts

```diff
@@ -78,32 +78,25 @@ export abstract class SimpleCustomResource {
     let cfnData: object | undefined;
 
     console.log(`Handling event: ${JSON.stringify(event)}`);
-    const requestType: string = event.RequestType;
     const resourceProperties: object = event.ResourceProperties ?? {};
     const physicalId: string = calculateSha256Hash(resourceProperties);
 
     try {
-      if (requestType === 'Create') {
-        if (!this.validateInput(resourceProperties)) {
-          throw Error(`Input did not pass validation check. Check log group "${context.logGroupName}" ` +
-            `for log stream ${context.logStreamName} for additional information.`);
-        }
-        cfnData = await this.doCreate(physicalId, resourceProperties);
-        console.debug(`Create data: ${JSON.stringify(cfnData)}`);
-      } else if (requestType === 'Update') {
-        if (!this.validateInput(resourceProperties)) {
-          throw Error('Input did not pass validation check');
-        }
-        const oldResourceProperties: object = event.OldResourceProperties ?? {};
-        const oldPhysicalId: string = calculateSha256Hash(oldResourceProperties);
-        if (oldPhysicalId !== physicalId) {
-          console.log('Doing Create -- ResourceProperties differ.');
-          cfnData = await this.doCreate(physicalId, resourceProperties);
-          console.debug(`Update data: ${JSON.stringify(cfnData)}`);
-        }
-      } else {
-        await this.doDelete(physicalId, resourceProperties);
-      }
+      const timeout = (prom: any, time: number, exception: any) => {
+        let timer: any;
+        return Promise.race([
+          prom,
+          new Promise((_r, rej) => timer = setTimeout(rej, time, exception)),
+        ]).finally(() => clearTimeout(timer));
+      };
+
+      // getRemainingTimeInMillis() should always be available. It is optional only in our code for backwards compatibility.
+      const remainingTime = context.getRemainingTimeInMillis?.() ?? 1000;
+      cfnData = await timeout(
+        this.handleEvent(event, context, resourceProperties, physicalId),
+        remainingTime - 1000,
```

Comment:
Is 1s enough time to reliably resolve the S3 put-object request that signals success/failure of the custom resource? Would we be better off with a timeout that's a bit more generous just in case there's a service blip in S3? e.g. 5s?

Reply:
The default Lambda timeout is actually 3 seconds. I could throw an exception with a message `Please increase Lambda timeout` if the `remainingTime` is less than 5 seconds. This will not give us a strong guarantee, but most likely will be caught during the development. What do you think?

Reply:
We could do some math maybe and do the warning for during development? Just to make sure that it'll still timeout and try to send the S3 response if the Lambda timeout is too low.
e.g. Something like `min(0.2 * remainingTime, 3000)` ?

Reply:
Sounds good. I've decided to keep 3 seconds instead of 5, used the math you proposed and also added printing a debug message if the remaining time is less than 3 seconds.